src/io/flutter/logging/FlutterLogFilterPanel.java

```diff
@@ -66,30 +71,39 @@ public boolean equals(Object o) {
     public int hashCode() {
       return Objects.hash(expression, isMatchCase, isRegex, logLevel);
     }
+
+    @Override
+    public String toString() {
+      return "FilterParam{" +
+             "expression='" + expression + '\'' +
+             ", isMatchCase=" + isMatchCase +
+             ", isRegex=" + isRegex +
+             ", logLevel=" + logLevel +
+             '}';
+    }
   }
 
   public interface OnFilterListener {
     void onFilter(@NotNull FilterParam param);
   }
 
-  @NotNull
-  private final OnFilterListener onFilterListener;
+  private static final OnFilterListener DEFAULT_ON_FILTER_LISTENER = param -> LOG.info("Requests filter: " + param);
+
+  @Nullable
+  private volatile OnFilterListener onFilterListener;
```

Comment:
I make `onFilterListener` as nullable because we have update preferences.
So while update preferences (user setting before) => all listener should not call.